src/libsync/account.cpp

```diff
@@ -302,37 +302,37 @@ void Account::setCapabilities(const Capabilities &caps)
     _capabilities = caps;
 }
 
-QString Account::serverVersion() const
+QString Account::serverProduct() const
 {
-    return _serverVersion;
+    return _serverProduct;
 }
 
-int Account::serverVersionInt() const
+QString Account::serverVersionString() const
 {
-    // FIXME: Use Qt 5.5 QVersionNumber
-    auto components = serverVersion().split(QLatin1Char('.'));
-    return makeServerVersion(components.value(0).toInt(),
-        components.value(1).toInt(),
-        components.value(2).toInt());
-}
-
-int Account::makeServerVersion(int majorVersion, int minorVersion, int patchVersion)
-{
-    return (majorVersion << 16) + (minorVersion << 8) + patchVersion;
+    return _serverVersion;
 }
 
 bool Account::serverVersionUnsupported() const
 {
-    if (serverVersionInt() == 0) {
+    if (serverVersion().isNull()) {
         // not detected yet, assume it is fine.
         return false;
     }
+    if (serverProduct() == QLatin1String("Infinite Scale")) {
+        return false;
+    }
     // Older version which is not "end of life" according to https://github.com/owncloud/core/wiki/Maintenance-and-Release-Schedule
-    return serverVersionInt() < makeServerVersion(10, 0, 0) || serverVersion().endsWith(QLatin1String("Nextcloud"));
+    return serverVersion() < QVersionNumber(10) || serverProduct().endsWith(QLatin1String("Nextcloud"));
+}
+
+QVersionNumber Account::serverVersion() const
+{
+    return QVersionNumber::fromString(_serverVersion);
 }
 
-void Account::setServerVersion(const QString &version)
+void Account::setServerVersion(const QString &version, const QString &productName)
```

Comment:
This setter's purpose has changed, so it should likely be renamed. There are two dedicated getters for both the version and the product name anyway.